lib/terminus.spec.js

```diff
@@ -160,16 +160,91 @@ describe('Terminus', () => {
     })
 
     it('returns 503 once signal received', (done) => {
-      execFile('node', ['lib/standalone-tests/terminus.onsignal.fail.js'])
+      let responseAssertionsComplete = false
+
+      // We're only truly finished when the response has been analyzed and the forked http process has exited,
+      // freeing up port 8000 for future tests
+      const finished = () => {
+        expect(responseAssertionsComplete).to.eql(true)
+        done()
+      }
+
+      execFile('node', ['lib/standalone-tests/terminus.onsignal.fail.js'], (error) => {
+        finished()
+
+        if (error) {
+          throw error
+        }
+      })
+
+      // let the process start up
+      setTimeout(() => {
+        fetch('http://localhost:8000/health')
+          .then(res => {
+            expect(res.status).to.eql(503)
+            responseAssertionsComplete = true
+          })
+      }, 300)
+    })
+
+    it('calls onSendFailureDuringShutdown when sending 503 during shutdown', (done) => {
+      let responseAssertionsComplete = false
+
+      // We're only truly finished when the response has been analyzed and the forked http process has exited,
+      // freeing up port 8000 for future tests
+      const finished = () => {
+        expect(responseAssertionsComplete).to.eql(true)
+        done()
+      }
+
+      execFile('node', ['lib/standalone-tests/terminus.onsendfailureduringshutdown.js'],
+        (error, stdout) => {
+          expect(stdout).to.eql('onSendFailureDuringShutdown\n')
+          finished()
+
+          if (error) {
+            throw error
+          }
+        })
+
+      // let the process start up
+      setTimeout(() => {
+        fetch('http://localhost:8000/health')
+          .then(res => {
+            expect(res.status).to.eql(503)
+            responseAssertionsComplete = true
+          })
+      }, 300)
+    })
+
+    it('does NOT call onSendFailureDuringShutdown when sending 503 during failed healthcheck', (done) => {
+      let responseAssertionsComplete = false
+
+      // We're only truly finished when the response has been analyzed and the forked http process has exited,
+      // freeing up port 8000 for future tests
+      const finished = () => {
+        expect(responseAssertionsComplete).to.eql(true)
+        done()
+      }
+
+      execFile('node', ['lib/standalone-tests/terminus.onsendfailureduringshutdown.failed-health.js'],
+        (error, stdout) => {
+          // Here, we expect NOT to see "onSendFailureDuringShutdown"
+          expect(stdout).to.eql('')
+          finished()
+
+          if (error) {
+            throw error
+          }
+        })
 
       // let the process start up
       setTimeout(() => {
         fetch('http://localhost:8000/health')
           .then(res => {
             expect(res.status).to.eql(503)
-            done()
+            responseAssertionsComplete = true
           })
-          .catch(done)
```

Comment:
With this catch in place, errors will cause the test to pass instead of fail. There needs to be only one call to `done()`, from the point where **all** of the assertions have been ran.